### DISCUSSION TITLE
Is queue support planned?

### DISCUSSION THREAD

**Discussion by DevJoghurt** [vercel/community#547] [Help] [answered]
Is queue support planned to allow running serverless functions via a queue in the background?

**Comment by styfle** [comment]
Hi @DevJoghurt 

There was some talk around a queue feature but nothing planned in the near future.

> In the docs one could still read until some time ago that a solution for this should come.

I don't recall a solution in the docs. Was this recommending a 3rd party?

**Comment by anthonyshort** [comment]
I was looking for this last night. I had to wire it up manually using SQS/SNS, which is painful. It'd be great if Vercel supported queues, pubsub, and cron for functions. You can wire up just about anything then.

**Reply by nhuethmayr** [reply]
Same here - 2 things that are missing are `Timed Functions` and `Message Queues`. For timed functions I'd recommend AWS Eventbrige (under the hood) and SQS for message queues. Firebase has both built in natively.

**Reply by AlessandroVol23** [reply]
Hi :) 

I had the exact same issue now a couple of times and want to build a product for that. I have some initial thoughts and a working prototype in a blog post [here](https://sandro.volpee.de/sqs-as-a-service)

**Comment by DevJoghurt** [comment]
@styfle 
Thanks for the answer.
I know it's an old version of the Docs but it was the information I had built my platform on:
https://docs-hcoknttm5.now.sh/docs/v2/advanced/platform/changes-in-now-2-0/

> No Daemons
Running a process persistently as a daemon requires a lot of operational overhead, such as setting up readiness checks, health checks and close monitoring.
>
>While 2.0 will not support this as a primitive, we still plan to address the following use-cases directly that our customers typically use processes for:
>
>Control and sync loops. A series of tasks that execute sequentially to verify or synchronize some remote state.
Scheduled jobs. E.g.: a task that needs to run every 50 minutes.
Jobs consumed from a queue.
We believe that these use-cases can be modeled and executed on much more effectively by taking full advantage of the serverless paradigm.
>
>To help our customers in the transition, we won't announce a deprecation date for 1.0 until these additional features are generally available.

Of course it's a pity, because I think the Zeit platform is really great and with queue support you could have really built everything with it.  But I can understand that the focus has changed now.  Then I guess I'll have to implement it with aws sam.

**Reply by coetry** [reply]
@DevJoghurt @anthonyshort Just echoing what @styfle said, we've discussed the topic and want to explore creating polished integrations with reliable queue providers. [CloudAMPQ](https://www.cloudamqp.com/) is a managed service and an alternative to SQS. 

Regarding pub/sub services like [Pusher](https://pusher.com) are excellent and have great DX.

**Reply by DevJoghurt** [reply]
The problem with integration is that you cannot develop locally with it.  But to be able to test and develop queues that trigger serverless functions locally, the whole thing would have to be part of the Zeit platform

**Reply by coetry** [reply]
Why wouldn't you be able to develop it locally? Integrations just make an easy way to configure the correct environment variables to get going with your project. Everything else after that should be the same.

**Comment by coetry** [comment]
Just echoing what @styfle said, we've discussed the topic and want to explore creating polished integrations with reliable queue providers. [CloudAMPQ](https://www.cloudamqp.com/) is a managed service and an alternative to SQS. 

Regarding pub/sub services like [Pusher](https://pusher.com) are excellent and have great DX.

**Reply by anthonyshort** [reply]
If integrations are the direction things are going, it'd be great to have docs explaining cron, queues, and pubsub, even if they're just explaining the best way to do it with Vercel and maybe provide some "blessed" examples.

**Comment by leerob** [comment, marked as answer]
**Edit: Update 2023**

Wanted to share an exciting update – we just released Vercel Cron Jobs!

Vercel Cron Jobs enable you to run scheduled jobs for automating backups and archiving, sending email and Slack notifications, and more. Cron Jobs can be used for any task you need to run on a schedule.

https://vercel.com/blog/cron-jobs

**Reply by nhuethmayr** [reply]
I believe @DevJoghurt was referring to a queuing system being built into Vercel without needing to "leave" the vercel platform and use AWS. Much like PubSub works on Firebase. 

Besides that, the article you've referred to, could mention using **AWS EventBridge** as a cron replacement. Its incredibly powerful, and for me, it feels right to stay within AWS (since Vercel also uses is).

**Comment by NazCodeland** [comment]
Is there any update on this?
